doc/modules/ROOT/pages/tutorials/kge-predict-transe-pyg-train.adoc

```diff
@@ -0,0 +1,469 @@
+        ttv_G, ["rel_id"], relationship_type, separate_property_columns=True
+    )
+    topology = [
+        rels_tmp.sourceNodeId.map(lambda x: nodeId_to_id[x]),
+        rels_tmp.targetNodeId.map(lambda x: nodeId_to_id[x]),
+    ]
+    edge_index = torch.tensor(topology, dtype=torch.long)
+    edge_type = torch.tensor(rels_tmp.rel_id.astype(int), dtype=torch.long)
+    data = Data(edge_index=edge_index, edge_type=edge_type)
+    data.num_nodes = len(nodeId_to_id)
+    display(data)
+    return data
+
+
+train_tensor_data = create_data_from_graph("TRAIN")
+test_tensor_data = create_data_from_graph("TEST")
+val_tensor_data = create_data_from_graph("VALID")
+----
+
+Drop the projected graph to save memory.
+
+[source, python, role=no-test]
+----
+gds.graph.drop(ttv_G)
+----
+
+The training process of the TransE model follows the corresponding PyG
+https://github.com/pyg-team/pytorch_geometric/blob/master/examples/kge_fb15k_237.py[example].
+
+[source, python, role=no-test]
+----
+def train_model_with_pyg():
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    model = TransE(
+        num_nodes=train_tensor_data.num_nodes,
+        num_relations=train_tensor_data.num_edge_types,
+        hidden_channels=50,
+    ).to(device)
+
+    loader = model.loader(
+        head_index=train_tensor_data.edge_index[0],
+        rel_type=train_tensor_data.edge_type,
+        tail_index=train_tensor_data.edge_index[1],
+        batch_size=1000,
+        shuffle=True,
+    )
+
+    optimizer = optim.Adam(model.parameters(), lr=0.01)
+
+    def train():
+        model.train()
+        total_loss = total_examples = 0
+        for head_index, rel_type, tail_index in loader:
+            optimizer.zero_grad()
+            loss = model.loss(head_index, rel_type, tail_index)
+            loss.backward()
+            optimizer.step()
+            total_loss += float(loss) * head_index.numel()
+            total_examples += head_index.numel()
+        return total_loss / total_examples
+
+    @torch.no_grad()
+    def test(data):
+        model.eval()
+        return model.test(
+            head_index=data.edge_index[0],
+            rel_type=data.edge_type,
+            tail_index=data.edge_index[1],
+            batch_size=20000,
+            k=10,
+        )
+
+    epoch_count = 501
+    for epoch in range(1, epoch_count):
+        loss = train()
+        print(f"Epoch: {epoch:03d}, Loss: {loss:.4f}")
+        if epoch % 75 == 0:
+            rank, hits = test(val_tensor_data)
+            print(f"Epoch: {epoch:03d}, Val Mean Rank: {rank:.2f}, " f"Val Hits@10: {hits:.4f}")
+
+    print(model)
+    rank, hits_at_10 = test(test_tensor_data)
+    print(f"Test Mean Rank: {rank:.2f}, Test Hits@10: {hits_at_10:.4f}")
+
+    torch.save(model, f"./model_{epoch_count}.pt")
+
+    return model
+----
+
+[source, python, role=no-test]
+----
+model = train_model_with_pyg()
+# The model can be loaded is it was trained before
```

Comment:
```suggestion
# The model can be loaded if it was trained before
```

Reply:
I'll align this change with original notebook text.

Reply:
@orazve Well spotted! I can change it for you and recreate the Asciidoc.